datastore/tieredds.nim

```diff
@@ -41,32 +41,40 @@ method has*(
 
 method delete*(
   self: TieredDatastore,
-  key: Key): Future[?!void] {.async.} =
+  key: Key): Future[?!void] {.async: (raises: [CancelledError]).} =
 
   let
     pending = await allFinished(self.stores.mapIt(it.delete(key)))
 
   for fut in pending:
-    if fut.read().isErr: return fut.read()
+    try:
+      if fut.read().isErr:
+        return fut.read()
+    except FuturePendingError as err:
+      return failure err
```

Comment:
For all of these blocks below, I don't think we really need to read the future, we can simply await it, and since they don't raise exceptions besides `CancelledError`, we don't need a try/except, eg:
```nim
for fut in pending:
  let res = await fut
  if res.isErr: return res
```